registry/reference.go

```diff
@@ -98,56 +98,92 @@ func ParseReference(artifact string) (Reference, error) {
 	}
 	registry, path := parts[0], parts[1]
 
+	var isTag bool
 	var repository string
 	var reference string
 	if index := strings.Index(path, "@"); index != -1 {
 		// `digest` found; Valid Form A (if not B)
+		isTag = false
```

Comment:
nit: This line is redundant since `isTag` is default to `false` at line 101.

Reply:
I get that it's redundant, but being explicit helps with readability (explicit communication of intent, vs accidental); that's why it's there.  Let me know if you think it's better removed though.